src/modules/ibmtts.c

```diff
@@ -1698,43 +1819,86 @@ static char *ibmtts_search_for_sound_icon(const char *icon_name)
 	return fn;
 }
 
-void alloc_voice_list()
+static void alloc_voice_list()
 {
-	enum ECILanguageDialect aLanguage[MAX_NB_OF_LANGUAGES];
-	int nLanguages = MAX_NB_OF_LANGUAGES;
+	enum ECILanguageDialect aLanguage[VOX_MAX_NB_OF_LANGUAGES];
+	int nLanguages = VOX_MAX_NB_OF_LANGUAGES;
 	int i = 0;
 
-	if (eciGetAvailableLanguages(aLanguage, &nLanguages))
+	/* if voxGetVoices available, update the list of installed voices */
+	if (_voxGetVoices) {
+		number_of_voices = VOX_RESERVED_VOICES;
+		if (!_voxGetVoices(voices, &number_of_voices) && (number_of_voices <= VOX_RESERVED_VOICES)) {
+			int i, j;
+			int min_id = eciLocales[VOX_ECI_VOICES-1].langID;			
+			for (i=0, j=VOX_ECI_VOICES; i<number_of_voices; i++) {
```

Comment:
It'd be better to avoid hardcoding that we will only have VOX_RESERVED_VOICES languages provided by voxin. That limit will be broken sooner or later.

Better:

* Keep `eciLocales` as it is in the current git master, just rename it to `eciHardcodedLocales`
* Introduce a global `eciLocale *eciLocales = eciHardcodedLocales;` pointer, which thus points by default to `eciHardcodedLocales`.
* When `voxGetvoices` is available, call it once with `NULL` to get the number of voices, then dynamically allocate a new `eciLocale` array of the proper size, and fill it.

That way, ibmtts will either use the hardcoded array (old voxin), or the dynamically-provided array (new voxin) without distorted array indexes etc.

All that being said, I don't understand why using `voxGetVoices`: AIUI voxin is just a forwarder for ibmtts, that is not actually necessary: the speech-dispatcher plugin can just directly link against ibmtts, and use ibmtts-provided functions. Here, won't the `eciGetAvailableLanguages` call simply just work already? If not yet, can't that be solved instead of introducing yet more complexity? AFAICT, `voxGetVoices` ends up just calling `eciGetAvailableLanguages` anyway, so I don't see the benefit of calling the former rather the latter, but see the drawback: it prevents from linking again ibmtts directly.